lib/Elastica/Dsn.php

```diff
@@ -0,0 +1,222 @@
+<?php
+
+namespace Elastica;
+
+use Elastica\Exception\InvalidException;
+
+/**
+ * Dns parser and configuration exporter to Elastica Client.
+ *
+ * @author Antoine Lamirault <lamiraultantoine@gmail.com>
+ */
+class Dsn
+{
+    /**
+     * @var string|null
+     */
+    private $host;
+
+    /**
+     * @var string|null
+     */
+    private $port;
+
+    /**
+     * @var string|null
+     */
+    private $url;
+
+    /**
+     * @var string|null
+     */
+    private $path;
+
+    /**
+     * @var string|null
+     */
+    private $proxy;
+
+    /**
+     * @var string|null
+     */
+    private $transport;
+
+    /**
+     * @var bool
+     */
+    private $persistent;
+
+    /**
+     * @var int|null
+     */
+    private $timeout;
+
+    /**
+     * @var array
+     */
+    private $connections;
+
+    /**
+     * @var bool
+     */
+    private $roundRobin;
+
+    /**
+     * @var bool
+     */
+    private $log;
+
+    /**
+     * @var int
+     */
+    private $retryOnConflict;
+
+    /**
+     * @var bool
+     */
+    private $bigintConversion;
+
+    /**
+     * @var string|null
+     */
+    private $username;
+
+    /**
+     * @var string|null
+     */
+    private $password;
+
+    /**
+     * Dsn constructor.
+     *
+     * @param string|null $host
+     * @param string|null $port
+     * @param string|null $url
+     * @param string|null $path
+     * @param string|null $proxy
+     * @param string|null $transport
+     * @param bool        $persistent
+     * @param int|null    $timeout
+     * @param array       $connections
+     * @param bool        $roundRobin
+     * @param bool        $log
+     * @param int         $retryOnConflict
+     * @param bool        $bigintConversion
+     * @param string|null $username
+     * @param string|null $password
+     */
+    public function __construct(?string $host, ?string $port, ?string $url, ?string $path, ?string $proxy,
+                                ?string $transport, bool $persistent, ?int $timeout, array $connections,
+                                bool $roundRobin, bool $log, int $retryOnConflict, bool $bigintConversion,
+                                ?string $username, ?string $password)
+    {
+        $this->host = $host;
+        $this->port = $port;
+        $this->url = $url;
+        $this->path = $path;
+        $this->proxy = $proxy;
+        $this->transport = $transport;
+        $this->persistent = $persistent;
+        $this->timeout = $timeout;
+        $this->connections = $connections;
+        $this->roundRobin = $roundRobin;
+        $this->log = $log;
+        $this->retryOnConflict = $retryOnConflict;
+        $this->bigintConversion = $bigintConversion;
+        $this->username = $username;
+        $this->password = $password;
+    }
+
+    public static function fromString(string $dsn): self
```

Comment:
As all params are kind of hardcoded, how do we keep it up-to-date on changes?

Reply:
What changes ? If you talk about config array maybe we have to move Dsn object to a Configuration object which have methods `fromDsn` and `fromArray` and use it in the Client.